iati/core/tests/test_rulesets.py

```diff
@@ -237,3 +238,118 @@ def rule_type(self):
     @pytest.fixture
     def valid_case(self):
         return {'paths': ['path_1', 'path_2']}
+
+    @pytest.fixture
+    def invalid_cases(self):
+        return {}
+
+
+class TestRuleDependent(RuleSubclassTestBase):
+    """A container for tests relating to RuleDependent."""
+
+    @pytest.fixture
+    def rule_type(self):
+        return 'dependent'
+
+    @pytest.fixture
+    def valid_case(self):
+        return {'paths': ['path_1', 'path_2']}
+
+    @pytest.fixture
+    def invalid_cases(self):
+        return {}
+
+
+class TestRuleSum(RuleSubclassTestBase):
+    """A container for tests relating to RuleSum."""
+
+    @pytest.fixture
+    def rule_type(self):
+        return 'sum'
+
+    @pytest.fixture
+    def valid_case(self):
+        return {'paths': ['path_1', 'path_2'], 'sum': 3}
+
+    @pytest.fixture(params=[{'paths': ['path_1', 'path_2']}, {'sum': 100}, {}])
+    def invalid_cases(self, request):
+        return request.param
+
+
+class TestRuleDateOrder(RuleSubclassTestBase):
+    """A container for tests relating to RuleDateOrder."""
+
+    @pytest.fixture
+    def rule_type(self):
+        return 'date_order'
+
+    @pytest.fixture
+    def valid_case(self):
+        return {'less': 'start', 'more': 'end'}
+
+    @pytest.fixture(params=[{'less': 'start'}, {'more': 'end'}, {}])
+    def invalid_cases(self, request):
+        return request.param
+
+
+class TestRuleRegexMatches(RuleSubclassTestBase):
+    """A container for tests relating to RuleRegexMatches."""
+
+    @pytest.fixture
+    def rule_type(self):
+        return 'regex_matches'
+
+    @pytest.fixture
+    def valid_case(self):
+        return {'regex': 'some regex', 'paths': ['path_1', 'path_2']}
+
+    @pytest.fixture(params=[{'regex': 'some regex'}, {'paths': ['path_1', 'path_2']}, {}])
```

Comment:
Likely worth testing a string that isn't a valid regex, such as `[` or `[0-9]++` ([source](https://stackoverflow.com/questions/7095238/an-invalid-regex-pattern))